src/App/App.js

```diff
@@ -24,35 +24,32 @@ class App extends Component {
       { error: 'Something went wrong, please try again later.' }))
   }
 
-  goBacktoMainView = () => {
-    this.setState({ selectedMovie: null });
-  }
-
-  handleClick = (id) => {
-    getMovieDetails(id)
-      .then(data => {
-        this.setState({ selectedMovie: data.movie })
-      })
-      .catch(() => this.setState(
-        { error: 'Something went wrong, please try again later.' }))
-  }
-
   render() {
     return (
       <div className="container">
         <main className='main'>
           {this.state.error &&
             <h1 className='error-message'>{this.state.error}</h1>}
-          {!this.state.selectedMovie && !this.state.error && <>
-            <Header />
-            <MovieList movies={ this.state.movies } handleClick={this.handleClick}/>
-          </>}
-          {!this.state.movies.length && !this.state.error && <>
+
+          {!this.state.movies.length && !this.state.error &&
             <h1 className='loading-message'>Movies are on the way!</h1>
-            </>}
-        </main>
-        {this.state.selectedMovie &&
-          <MovieDetail movie={this.state.selectedMovie} goBack={this.goBacktoMainView}/>}
+            }
+        <Switch>
+         <Route exact path='/' render={() =>
+           { return ( <>
+              <Header />
+              <MovieList movies={ this.state.movies } />
+            </>
+           )
+          }
+          }/>
+        <Route path='/:id' render={({ match }) => {
+          const { id } = match.params
+           return <MovieDetail id={ parseInt(id) } />
+        }
+        }/>
+          </Switch>
+         </main>
       </div>
```

Comment:
There seem to be some formatting/spacing issues line 51, 52, and 53- it looks like they could all be bumped back one or two spaces.